### PR TITLE
[Internal] Refine the retry-after interval for openai retry error

### PR DESCRIPTION
**Pull request by dorisjoy**
# Description

Currently the retry-after interval for openai retry error is multiplied by (2 ** i):
1. error without retry-after header: 8 * (2 ** i);
2. error with retry-after header: float(retry_after_in_header) * (2 ** i)
This calculation will cause the retry-after very large when the i (retry_count) is large, e.g.: 608s
image

This will result in:
1. the job running for a long time, and most of the time are waiting interval, make the resource been wasted.
2. Some lines will failed with timeout because of the long retry-after interval.

Thus, we want to refine the retry-after interval for openai retry error like below:
1. error without retry-after header: minbackoffinsec + (2**i - 1). If this value is larger than 60s, then use 60s instead.
2. error with retry-after header: float(retry_after_in_header). Just respect the retry-after value returned from openai.

Test:
1. Create a test package for this change: 
pip install promptflow-tools==0.0.119298968 --extra-index-url https://azuremlsdktestpypi.azureedge.net/test-promptflow/
3. Create a runtime with this test package, and below is the test result:
original link:
https://ml.azure.com/prompts/flow/f8925d22-478a-4b5c-8e16-a85158d03b9b/bf1c7abc-212e-4544-bcb8-3fce3f0ae627/run/9b6d8fe7-f2da-4328-a454-c2467ecd2db5/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourceGroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus-dev&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
55 out of 1000 lines has failed because of 10 mins timeout;
link with fix:
https://ml.azure.com/prompts/flow/f8925d22-478a-4b5c-8e16-a85158d03b9b/bf1c7abc-212e-4544-bcb8-3fce3f0ae627/run/1491c004-cc6c-475e-8c75-76b4c9b595e4/details?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourceGroups/promptflow/providers/Microsoft.MachineLearningServices/workspaces/promptflow-eastus-dev&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
all 1000 lines has succeeded.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-tools/promptflow/tools/common.py [hunk 1]
```diff
@@ -178,7 +178,18 @@ def to_content_str_or_list(chat_str: str, hash2images: Mapping):
     return result if include_image else chat_str
 
 
-def handle_openai_error(tries: int = 10, delay: float = 8.0):
+def generate_retry_interval(retry_count: int) -> float:
+    minBackoffInSec = 3
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/common.py:182]: better to use snake case for variable name?

File: src/promptflow-tools/promptflow/tools/common.py [hunk 2]
```diff
@@ -178,7 +178,18 @@ def to_content_str_or_list(chat_str: str, hash2images: Mapping):
...
+    maxBackoffInSec = 60
+    retry_interval = minBackoffInSec + ((2 ** retry_count) - 1)
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/common.py:184]: Just curious why minus 1?

File: src/promptflow-tools/promptflow/tools/common.py (hunk 1 above)
Comment (dorisjoy) [reply src/promptflow-tools/promptflow/tools/common.py:182]: changed.

File: src/promptflow-tools/promptflow/tools/common.py (hunk 2 above)
Comment (dorisjoy) [reply src/promptflow-tools/promptflow/tools/common.py:184]: For the first retry, we want it to use the min_backoff_in_sec (3s) directly. And here the retry_count is 0, so use 2**retry_count - 1 to achieve " + 0".

**Review by chjinche — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Review by DaweiCai — APPROVED** [review]

**Pull request merged**